pkg/controller/disruption/disruption.go

```diff
@@ -783,9 +804,46 @@ func (dc *DisruptionController) updatePdbStatus(pdb *policy.PodDisruptionBudget,
 		ObservedGeneration: pdb.Generation,
 	}
 
+	if newPdb.Status.Conditions == nil {
+		newPdb.Status.Conditions = make([]metav1.Condition, 0)
+	}
+	if newPdb.Status.DisruptionsAllowed > 0 {
+		apimeta.SetStatusCondition(&newPdb.Status.Conditions, metav1.Condition{
+			Type:               DisruptionAllowedCondition,
+			Reason:             SufficientPodsReason,
+			Status:             metav1.ConditionTrue,
+			ObservedGeneration: newPdb.Status.ObservedGeneration,
+		})
+	} else {
+		apimeta.SetStatusCondition(&newPdb.Status.Conditions, metav1.Condition{
+			Type:               DisruptionAllowedCondition,
+			Reason:             InsufficientPodsReason,
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: newPdb.Status.ObservedGeneration,
+		})
+	}
+
 	return dc.getUpdater()(newPdb)
 }
 
+func conditionsAreUpToDate(pdb *policy.PodDisruptionBudget) bool {
+	cond := apimeta.FindStatusCondition(pdb.Status.Conditions, DisruptionAllowedCondition)
+	if cond == nil {
+		return false
+	}
+
+	if pdb.Status.DisruptionsAllowed > 0 {
+		if cond.Status == metav1.ConditionTrue && cond.Reason == SufficientPodsReason {
+			return true
+		}
+	} else {
+		if cond.Status == metav1.ConditionFalse && cond.Reason == InsufficientPodsReason {
+			return true
+		}
+	}
+	return false
```

Comment:
```suggestion
	return cond.Status == metav1.ConditionFalse && cond.Reason == InsufficientPodsReason
```

Reply:
Yeah, this is clearer. I have updated the PR with these changes.